### ISSUE TITLE
Make the clients initialize the servers themself

### ISSUE THREAD

**Issue by saccarosium** [Olical/conjure#625]
Hi,
I think we can have a mechanism that lets the various clients initialize the servers for the user.

Taking as an example guile via socket. It is annoying that I have to spin up myself the repl and set the socket. It would be much nicer I do `<localleader>cc` and then the guile process is spawned by neovim via a job and sets ups the socket for me. I think it can be really easy to make and we can use just a function that can be user customize if needed.

**Comment by Olical** [comment]
Yep this is a per client thing, it's also a per user thing. If I turned this on by default for all clients I'm sure it'd make some people happy and some very unhappy, so we'll need more options and documentation for something like this.

Maybe a global option that all clients should try to respect around auto starting / connecting to REPLs. This is now on my radar, but will need a little thought about how to do it nicely.

Maybe having all clients respect a global option is a good idea, try to unify them into a common behaviour. It'll just be through convention so will require changes in every client which is a bit time consuming.

**Comment by russtoku** [comment]
In v4.39.0, a global option was added to disable automatically calling a client's `on_load` function.
> Add option to disable client on-load fns https://github.com/Olical/conjure/issues/377
on-load is used by clients to trigger auto repls and connections. This
means you can make Conjure a little more passive until you invoke
something that requires a connection / repl.

I implemented code for the Python, snd-7, and SQL clients so that they wouldn't start the REPL connection if `g:conjure#client_on_load` was set to `v:false`. Unfortunately, I mentioned this only in the help docs for the SQL client.

It doesn't seem like other clients have implemented code that enables this functionality.

Similarly, I think that each client should handle automatically launching their REPL server on their own. Providing a function to do that and `<localleader>cc` to execute that function would be nice. Other clients could borrow parts or all of the implementation from the Guile client if needed.
